### PR TITLE
adding inProduct Feedback to Enterprise Apps API

### PR DESCRIPTION
**Pull request by luleonpla**

### CODE REVIEW

**Comment by luleonpla** [comment]
@bettirosengugi I'm getting some validation errors about humanName not being lower case but the errors are referring to other APIs descriptions. Also, recently I got a conflict for a file that I didn't modify. Could you help here?

**Comment by irvinesunday** [comment]
> @bettirosengugi I'm getting some validation errors about humanName not being lower case but the errors are referring to other APIs descriptions. Also, recently I got a conflict for a file that I didn't modify. Could you help here?

Thank you for your contribution @luleonpla. Please ignore the `humanName` validation errors if they are from other descriptions. There is a separate work to get those fixed, and they haven't been merged in yet, that's why you're seeing the errors. About the conflict file, could you try to merge the `dev` branch onto your branch (and accept the changes from the `dev` branch). That should help clear that conflict.

**Comment by luleonpla** [comment]
Thank you @irvinesunday. I fixed the conflict. So, this should be ready for your review.

File: sample-queries/sample-queries.json [hunk 1]
```diff
@@ -1981,6 +1981,76 @@
...
         "skipTest": false
+    },
+    {
+        "id": "fd58d371-315e-49d8-aa74-58fc4ad1337c",
+        "category": "Identity and Access",
+        "method": "GET",
+        "humanName": "list Azure ad gallery app",
+        "requestUrl": "/beta/applicationTemplates",
+        "docLink": "https://docs.microsoft.com/graph/api/applicationtemplate-list?view=graph-rest-beta",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/appTemplateAPISurvey",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:1992]: @bettirosengugi - thoughts on adding the `feedback` message on the `tip` property?

File: sample-queries/sample-queries.json [hunk 2]
```diff
@@ -1981,6 +1981,76 @@
...
+        "requestUrl": "/v1.0/servicePrincipals/{id}",
+        "docLink": "https://docs.microsoft.com/graph/api/serviceprincipal-update?view=graph-rest-1.0",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/servicePrincipalAPISurvey",
+        "skipTest": false
+    },
+    {
+        "id": "42d5a2bc-4376-4b40-bbb5-dfad82879fff",
+        "category": "Identity and Access",
+        "method": "PATCH",
+        "humanName": "update properties on application",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:2019]: Nitpick: Update the properties of an application

File: sample-queries/sample-queries.json [hunk 3]
```diff
@@ -1981,6 +1981,76 @@
...
+        "humanName": "list Azure ad gallery app",
+        "requestUrl": "/beta/applicationTemplates",
+        "docLink": "https://docs.microsoft.com/graph/api/applicationtemplate-list?view=graph-rest-beta",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/appTemplateAPISurvey",
+         "skipTest": false
+    },
+    {
+        "id": "101af4d8-4697-4347-906e-9c3659bf2a29",
+        "category": "Identity and Access",
+        "method": "POST",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:1998]: Let's add `postBody` values for the `POST` and `PATCH` samples.

File: sample-queries/sample-queries.json (hunk 1 above)
Comment (bettirosengugi) [reply sample-queries/sample-queries.json:1992]: This is fine.

**Review by bettirosengugi — APPROVED** [review]

**Comment by luleonpla** [comment]
@bettirosengugi @irvinesunday I have addressed all the comments. Let me know if we're good to merge the PR.

File: sample-queries/sample-queries.json [hunk 4]
```diff
@@ -1981,6 +1981,80 @@
...
+    },
+    {
+        "id": "faf23cd6-070e-4c98-bbba-61c7d73163cc",
+        "category": "Identity and Access",
+        "method": "POST",
+        "humanName": "create a claim mapping policy",
+        "requestUrl": "/v1.0/policies/claimsMappingPolicies",
+        "docLink": "https://docs.microsoft.com/graph/api/claimsmappingpolicy-post-claimsmappingpolicies?view=graph-rest-1.0",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/claimsMappingPolicyAPISurvey",
+        "postBody":"{\r\n  \"definition\": [\r\n    \"definition-value\"\r\n  ],\r\n  \"displayName\": \"displayName-value\",\r\n  \"isOrganizationDefault\": true,\r\n  \"id\": \"id-value\"\r\n}",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:2034]: The `postBody` for the [request ](https://docs.microsoft.com/en-us/graph/api/claimsmappingpolicy-post-claimsmappingpolicies?view=graph-rest-1.0&tabs=http#request)doesn't have an `id-value` property. It's the [response ](https://docs.microsoft.com/en-us/graph/api/claimsmappingpolicy-post-claimsmappingpolicies?view=graph-rest-1.0&tabs=http#response-1)that has that property.

File: sample-queries/sample-queries.json [hunk 5]
```diff
@@ -1981,6 +1981,80 @@
...
+        "requestUrl": "/beta/applicationTemplates/{id}/instantiate",
+        "docLink": "https://docs.microsoft.com/graph/api/applicationtemplate-instantiate?view=graph-rest-beta",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/appTemplateAPISurvey",
+        "postBody": "{\r\n \"displayName\": \"appDisplayName\"\r\n}",
+        "skipTest": false
+    },
+    {
+        "id": "3155cff4-2d16-4d45-a9f7-214b6a246dc5",
+        "category": "Identity and Access",
+        "method": "PATCH",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:2009]: We can add a sample request `postBody` with just a single property as supplied in the [docs](https://docs.microsoft.com/en-us/graph/api/serviceprincipal-update?view=graph-rest-1.0&tabs=http#request); to help our users with the initial request construction.

File: sample-queries/sample-queries.json [hunk 6]
```diff
@@ -1981,6 +1981,80 @@
...
+        "humanName": "update properties on the servicePrincipal",
+        "requestUrl": "/v1.0/servicePrincipals/{id}",
+        "docLink": "https://docs.microsoft.com/graph/api/serviceprincipal-update?view=graph-rest-1.0",
+        "tip": "We’d like to hear from you. Please leave your feedback on this API here: https://aka.ms/servicePrincipalAPISurvey",
+        "skipTest": false
+    },
+    {
+        "id": "42d5a2bc-4376-4b40-bbb5-dfad82879fff",
+        "category": "Identity and Access",
+        "method": "PATCH",
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:2019]: Same as above.

File: sample-queries/sample-queries.json (hunk 5 above)
Comment (luleonpla) [reply sample-queries/sample-queries.json:2009]: Added.

File: sample-queries/sample-queries.json (hunk 6 above)
Comment (luleonpla) [reply sample-queries/sample-queries.json:2020]: added

File: sample-queries/sample-queries.json (hunk 5 above)
Comment (luleonpla) [reply sample-queries/sample-queries.json:2009]: Added the postBody

File: sample-queries/sample-queries.json (hunk 4 above)
Comment (luleonpla) [reply sample-queries/sample-queries.json]: Removed id

**Comment by luleonpla** [comment]
@irvinesunday I've updated the files with the feedback. Please review.

**Comment by luleonpla** [comment]
@irvinesunday @bettirosengugi can we move forward and merge this PR? I think I have addressed all the feedback. Thanks.

**Comment by sanana30** [comment]
Hi @thewahome - can you review this request? I see that the pull request is still open. Thanks!

CC: @luleonpla @bettirosengugi

**Review by irvinesunday — APPROVED** [review]
👍🏽

**Pull request merged**